libs/qCC_io/AsciiFilter.cpp

```diff
@@ -845,7 +846,10 @@ CC_FILE_ERROR AsciiFilter::loadCloudFromFormatedAsciiFile(	const QString& filena
 					ccLog::Error("Not enough memory! Process stopped ...");
 					break;
 				}
-				cloudDesc.cloud->setGlobalShift(Pshift);
+				if (preserveCoordinateShift)
```

Comment:
Instead of all the extra `if (preserveCoordinateShift)` code around setting the global shift everywhere, wouldn't it be cleaner to add a boolean to `ccShiftedObject`?

Than change `setGlobalShift()` to take & set the bool?

Reply:
I would find funny to keep this information all the way for nothing... But it is clearly a viable option :D

Reply:
It's a bool per cloud, right? Seems like a reasonable trade-off if it clean any of this up :-)